query-engine/core/src/interactive_transactions/manager.rs

```diff
@@ -109,24 +109,37 @@ impl ItxManager {
         isolation_level: Option<String>,
         timeout: Duration,
     ) -> crate::Result<()> {
-        // This task notifies the task spawned in `new()` method that the timeout for this
-        // transaction has expired.
-        crosstarget_utils::task::spawn({
-            let timeout_sender = self.timeout_sender.clone();
-            let tx_id = tx_id.clone();
-            async move {
-                crosstarget_utils::time::sleep(timeout).await;
-                timeout_sender.send(tx_id).expect("receiver must exist");
+        // Only create a client if there is no client for this transaction yet.
+        // otherwise, begin a new transaction/savepoint for the existing client.
+        if self.transactions.read().await.contains_key(&tx_id) {
+            let transaction_entry = self.get_transaction(&tx_id, "begin").await?;
+            let mut tx = transaction_entry.lock().await;
+            // If the transaction is already open, we need to create a savepoint.
+            if tx.depth() > 0 {
+                tx.create_savepoint().await?;
+            } else {
+                tx.begin().await?;
```

Comment:
Is it possible/expected for this code branch to execute?

Reply:
@aqrln Reading back through the code, this shouldn't ever be hit. If the transaction depth is 0 or lower, then the transaction should be committed or rolled back.

Reply:
Resolved 👍